cli/azd/pkg/project/scaffold_gen.go

```diff
@@ -270,21 +228,97 @@ func infraSpec(projectConfig *ProjectConfig) (*scaffold.InfraSpec, error) {
 	return &infraSpec, nil
 }
 
-func mapContainerApp(res *ResourceConfig, svcSpec *scaffold.ServiceSpec, infraSpec *scaffold.InfraSpec) error {
-	props := res.Props.(ContainerAppProps)
+func mapUses(infraSpec *scaffold.InfraSpec, projectConfig *ProjectConfig) error {
```

Comment:
In the new function, it handles all `uses` in one place for both backing services (like mysql) and container app (like `web` uses `api`)